server/errors/api-error.js

```diff
@@ -0,0 +1,23 @@
+const AbstractError = require('../../abstract-error')
+const httpStatus = require('http-status')
+
+/**
+ * Class representing an API error.
+ * @extends AbstractError
+ */
+class ApiError extends AbstractError {
+  /**
+   * Creates an API error.
+   * @param {string} message - Error message.
+   * @param {number} status - HTTP status code.
+   * @param {boolean} isPublic - Whether the error message should be public.
+   */
+  constructor (message, status = httpStatus.INTERNAL_SERVER_ERROR,
```

Comment:
Sure! You're right! Sorry! One thing came to my mind... 500 family will not be thrown by ApiError, only the 400 family. So I think 400 (BAD_REQUEST) should be the default status error.

INTERNAL_SERVER_ERROR will occur unexpectedly. This is handled by the uncaught error handler.